rizinorg/rizin
# Fix aae overwriting registers of debuggee

**Your checklist for this pull request**

**Detailed description**

Running `aaa` (in particular its `aae`) part would mess up the register contents of the debuggee.
The changes in tests are because some of the internal effects of analysis are not persisted anymore.

**Test plan**

Since this depends on debug specifically, it's not that easy to automate testing:

* Install https://git.sr.ht/~thestr4ng3r/rz-commodore
* Start `x64sc` with binary remote monitor enabled
* Open `rz -a 6502 -d vice://`
* Note the output of `dr`
* Run `aaa`
* Run `dr` again. The contents should not have changed compared to before the analysis.
* Run `dc` and observe that the debuggee does not get stuck or do any other weird stuff.
